source/core/ut_suite_builder.pkb

```diff
@@ -428,39 +428,40 @@ create or replace package body ut_suite_builder is
 
   procedure add_test(
     a_suite            in out nocopy ut_suite,
-    a_tests            in out nocopy tt_tests,
+    a_suite_items      in out nocopy ut_suite_items,
     a_procedure_name   t_object_name,
-    a_proc_annotations tt_annotations_by_name
+    a_annotations      t_annotations_info
   ) is
     l_test             ut_test;
     l_annotation_texts tt_annotation_texts;
-    l_annotation_pos   binary_integer;
+    l_proc_annotations tt_annotations_by_name :=  a_annotations.by_proc(a_procedure_name);
   begin
-    if not a_proc_annotations.exists(gc_test) then
+    if not l_proc_annotations.exists(gc_test) then
       return;
     end if;
-    warning_on_duplicate_annot(a_suite, a_proc_annotations, gc_test, a_procedure_name);
-    warning_on_duplicate_annot(a_suite, a_proc_annotations, gc_displayname, a_procedure_name);
-    warning_on_duplicate_annot(a_suite, a_proc_annotations, gc_rollback, a_procedure_name);
+    warning_on_duplicate_annot( a_suite, l_proc_annotations, gc_test, a_procedure_name);
+    warning_on_duplicate_annot( a_suite, l_proc_annotations, gc_displayname, a_procedure_name);
+    warning_on_duplicate_annot( a_suite, l_proc_annotations, gc_rollback, a_procedure_name);
     warning_bad_annot_combination(
-        a_suite, a_procedure_name, a_proc_annotations, gc_test,
+        a_suite, a_procedure_name, l_proc_annotations, gc_test,
         ut_varchar2_list(gc_beforeeach, gc_aftereach, gc_beforeall, gc_afterall)
     );
-    
-    l_test := ut_test(a_suite.object_owner, a_suite.object_name, a_procedure_name);
 
-    if a_proc_annotations.exists(gc_displayname) then
-      l_annotation_texts := a_proc_annotations(gc_displayname);
+    l_test := ut_test(a_suite.object_owner, a_suite.object_name, a_procedure_name, l_proc_annotations( gc_test).first);
+    l_test.parse_time  := a_annotations.parse_time;
+
+    if l_proc_annotations.exists( gc_displayname) then
+      l_annotation_texts := l_proc_annotations( gc_displayname);
       --take the last definition if more than one was provided
       l_test.description := l_annotation_texts(l_annotation_texts.first);
       --TODO if more than one - warning
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this "TODO" comment. ([plsql:S1135](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=783&issues=AWcfCxOld6oxh3R4qk8H&open=AWcfCxOld6oxh3R4qk8H)

[ISSUE_KEY]: AWcfCxOld6oxh3R4qk8H